src/findings_data_import.py

```diff
@@ -49,12 +49,292 @@ def move_processed_file(s3_client, bucket, folder, filename):
+        ):
+            response = ssm_client.get_parameter(
+                Name=ssm_param_name, WithDecryption=True
+            )
+            db_info[key] = response["Parameter"]["Value"]
+
+        logging.info("Connecting to the mongo db at %s %s", db_hostname, db_port)
+        # Set up database connection
+        credPw = urllib.parse.quote(db_info["password"])
+        db_uri = (
+            f"mongodb://{db_info['username']}:{credPw}@"
+            f"{db_hostname}:{db_port}/{db_info['db_name']}"
+        )
+
+        # Connect to MongoDB with timeout so Lambda doesn't run over
+        db_connection = MongoClient(
+            host=db_uri, serverSelectionTimeoutMS=2500, tz_aware=True
+        )
+        db = db_connection[db_info["db_name"]]
+        logging.info(
+            "DB connection set up to %s:%s/%s", db_hostname, db_port, db_info["db_name"]
+        )
+        return db
+    except ClientError as client_err:
+        raise Exception(
+            "Unable to fetch database credentials from the SSM", client_err
+        ).with_traceback()
+    # Handle all mongo exceptions the same way..
+    except Exception as err:
+        raise Exception("Unable to connect to the mongo db", err).with_traceback()
+
+
+def download_file(
+    s3_client=None,
+    s3_bucket=None,
+    data_filename=None,
+):
+    """Download a file from a specified s3 bucket, and place it in a temporary file path.
+
+    Parameters:
+    -----------
+    s3_client : S3.Client
+        The AWS S3 Client to retrieve the data file with.
+
+    s3_bucket : str
+        The AWS S3 bucket containing the data file.
+
+    data_filename : str
+        The name of the file containing the data in the S3 bucket
+        above.
+
+    Returns
+    -------
+        string : The file path of the newly created temp file
+        dict   : The JSON data dictionary loaded from downloaded file
+    """
+    logging.info("Retrieving %s from %s...", data_filename, s3_bucket)
+
+    try:
+        # Securely create a temporary file to store the JSON data in
+        temp_file_descriptor, temp_data_filepath = tempfile.mkstemp()
+        # Fetch findings data file from S3 bucket
+        s3_client.download_file(
+            Bucket=s3_bucket, Key=data_filename, Filename=temp_data_filepath
+        )
+
+        # Load data JSON
+        with open(temp_data_filepath) as data_json_file:
+            findings_data = json.load(data_json_file)
+
+        logging.info("JSON data loaded from %s.", data_filename)
+        return temp_data_filepath, findings_data
+    except json.JSONDecodeError as json_err:
+        raise (
+            Exception(
+                f"Unable to decode JSON data for {data_filename}", json_err
+            ).with_traceback()
+        )
+    except ClientError as err:
+        raise (
+            Exception(
+                f"Error downloading file {data_filename} from {s3_bucket} ", err
+            ).with_traceback()
+        )
```

Comment:
Same exception handling rationale as above.

```suggestion
    except json.JSONDecodeError:
        logging.error("Unable to decode JSON data for %s", data_filename)
        raise
    except ClientError:
        logging.error("Error downloading file %s from %s", data_filename, s3_bucket)
        raise
```